nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ListenSMTP.java

```diff
@@ -166,317 +108,158 @@
-        final SMTPServer server = new SMTPServer(smtpMessageHandlerFactory) {
-
+    /**
+     *
+     */
+    private SMTPServer createServerInstance(ProcessContext context, SmtpConsumer consumer) {
+        SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+        SMTPServer smtpServer = sslContextService == null ? new SMTPServer(consumer) : new SMTPServer(consumer) {
             @Override
             public SSLSocket createSSLSocket(Socket socket) throws IOException {
                 InetSocketAddress remoteAddress = (InetSocketAddress) socket.getRemoteSocketAddress();
-
-                SSLSocketFactory socketFactory = finalSslContext.getSocketFactory();
-
-                SSLSocket s = (SSLSocket) (socketFactory.createSocket(socket, remoteAddress.getHostName(), socket.getPort(), true));
-
-                s.setUseClientMode(false);
-
-
-                // For some reason the createSSLContext above is not enough to enforce
-                // client side auth
-                // If client auth is required...
-                if (SSLContextService.ClientAuth.REQUIRED.toString().equals(context.getProperty(CLIENT_AUTH).getValue())) {
-                    s.setNeedClientAuth(true);
+                String clientAuth = context.getProperty(CLIENT_AUTH).getValue();
+                SSLContext sslContext = sslContextService.createSSLContext(SSLContextService.ClientAuth.valueOf(clientAuth));
+                SSLSocketFactory socketFactory = sslContext.getSocketFactory();
+                SSLSocket sslSocket = (SSLSocket) (socketFactory.createSocket(socket, remoteAddress.getHostName(),socket.getPort(), true));
+                sslSocket.setUseClientMode(false);
+
+                if (SSLContextService.ClientAuth.REQUIRED.toString().equals(clientAuth)) {
+                    this.setRequireTLS(true);
+                    sslSocket.setNeedClientAuth(true);
                 }
 
-
-                return s;
+                return sslSocket;
             }
         };
-
-        // Set some parameters to our server
-        server.setSoftwareName("Apache NiFi");
-
-
-        // Set the Server options based on properties
-        server.setPort(context.getProperty(SMTP_PORT).asInteger());
-        server.setHostName(context.getProperty(SMTP_HOSTNAME).getValue());
-        server.setMaxMessageSize(context.getProperty(SMTP_MAXIMUM_MSG_SIZE).asDataSize(DataUnit.B).intValue());
-        server.setMaxConnections(context.getProperty(SMTP_MAXIMUM_CONNECTIONS).asInteger());
-        server.setConnectionTimeout(context.getProperty(SMTP_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue());
-
-
-        // Check if TLS should be enabled
         if (sslContextService != null) {
-            server.setEnableTLS(true);
-        } else {
-            server.setHideTLS(true);
```

Comment:
@olegz TLS should be hidden when not enabled so to prevent situations where the client will not like the `454 TLS not supported` message ListenSMTP returns when it is disabled.

This is required to prevent this:

```
$ telnet 0 2525
Trying 0.0.0.0...
Connected to 0.
Escape character is '^]'.
220 localhost ESMTP Apache NiFi
help
214-Apache NiFi on localhost
214-Topics:
214-     HELP
214-     HELO
214-     RCPT
214-     MAIL
214-     DATA
214-     AUTH
214-     EHLO
214-     NOOP
214-     RSET
214-     VRFY
214-     QUIT
214-     STARTTLS
214-For more info use "HELP <topic>".
214 End of HELP info
STARTTLS
454 TLS not supported
quit
221 Bye
Connection closed by foreign host.
```

vs

```
$ time openssl s_client -starttls smtp -crlf -connect 127.0.0.1:2525
CONNECTED(00000003)
didn't found starttls in server response, try anyway...
140406973187912:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protocol:s23_clnt.c:769:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 119 bytes and written 282 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
---

real    1m0.086s
user    0m0.023s
sys     0m0.003s

```